.github/workflows/wheel_tests_and_release.yml

```diff
@@ -48,6 +45,14 @@ jobs:
           python -m build --no-isolation --skip-dependency-check --sdist .
           ls -la ${{ github.workspace }}/dist
 
+      - name: Verify the distribution
+        run: pipx run twine check --strict dist/*
+
+      - name: Generate artifact attestation for sdist and wheels
+        uses: actions/attest-build-provenance@173725a1209d09b31f9d30a3890cf2757ebbff0d # v1.1.2
+        with:
+          subject-path: "dist/scikit_image-*"
```

Comment:
c.f. https://github.com/actions/attest-build-provenance for more information.

Once this runs during a release the attestations will be uploaded to https://github.com/scikit-image/scikit-image/attestations and can be verified from a wheel or sdist artifact using the `gh attestation verify` CLI API. c.f. https://github.com/scikit-hep/pyhf/pull/2473 for examples of that.